openedx/features/redhouse_features/management/commands/sync_users_data.py

```diff
@@ -0,0 +1,137 @@
+
+DEFAULT_SUB_ORGANIZATION = 'redhouse'
+DEFAULT_ORGANIZATION = 'redhouse'
+
+
+class Command(BaseCommand):
+    """
+    Sync users data according to multisites requirements
+
+    This command MUST be run in cms.
+
+    Format:
+        python manage.py cms sync_users_data [sub_organization [organization]]
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('sub_organization', nargs='?', default=DEFAULT_SUB_ORGANIZATION, type=str)
+        parser.add_argument('organization', nargs='?', default=DEFAULT_ORGANIZATION, type=str)
+
+    def handle(
+        self, *args, **options
+    ):
+        sub_organization = options['sub_organization']
+        organization = options['organization']
+
+        users = User.objects.filter(is_superuser=False)
+        staff_users = users.filter(is_staff=True)
+
+        sub_org = self.get_sub_organization(sub_organization)
+
+        linked_users_count = len(
+            [
+                1 for user in users if self.create_and_link_edly_profiles(user, sub_org)
+            ]
+        )
+
+        course_creator_count = len([
+            1 for user in staff_users if self.create_course_creator(user)
+        ])
+        course_access_roles_count = len([
+            1 for user in staff_users if self.create_course_access_roles(user, organization)
+        ])
+
+        map(self.remove_staff_access, staff_users)
+
+        log.info('Edly user profiles are successfully created for %d user(s)', linked_users_count)
+        log.info('Course creator objects are successfully created for %d user(s)', course_creator_count)
+        log.info('Course access roles are successfully created for %d user(s)', course_access_roles_count)
+        log.info('Staff access is removed for %d user(s)', course_access_roles_count)
+
+    def get_sub_organization(self, sub_org_slug):
+        try:
+            return EdlySubOrganization.objects.get(slug=sub_org_slug)
+        except EdlySubOrganization.DoesNotExist:
+            log.error('Edly sub organization %s does not exist', sub_org_slug)
+            exit(1)
+
+    def create_and_link_edly_profiles(self, user, sub_org):
+        try:
+            edly_profile = EdlyUserProfile.objects.get(user=user)
+            log.info(
+                'Found existing edly user profile %s linked with sub org %s',
+                user.username,
+                sub_org
+            )
+            return
+        except EdlyUserProfile.DoesNotExist:
+            edly_profile = EdlyUserProfile.objects.create(user=user)
+            edly_profile.edly_sub_organizations.add(sub_org)
+            edly_profile.save()
+            log.info(
+                'Created edly user profile for user %s and linked with sub org %s',
+                user.username,
+                sub_org
+            )
+            return edly_profile
+
+    def create_course_creator(self, user):
+        try:
+            course_creator = CourseCreator.objects.get(user=user)
```

Comment:
What if the user is present in CourseCreator but hasn't granted the access yet?

Reply:
As mentioned above, its very unlikey, but if this is the case, that object will be handled manually.